spec/models/provider_spec.rb

```diff
@@ -37,90 +46,113 @@ module Isomorphic
     end
   end
 
-  describe '.all' do
+  describe '#initialize' do
+
+    it 'sets given attributes' do
+      expect(described_class.new({ rdf_subject: rdf_subject }).rdf_subject)
+        .to eq rdf_subject
+    end
+  end
+
+  describe '#all' do
     it 'with no items is empty' do
       expect(described_class.all).to be_empty
     end
 
-    context 'with item' do
-      include_context 'with indexed item'
+    context 'with valid item' do
+      include_context 'indexed in Solr'
 
-      it 'returns all items' do
-        # todo: ActiveTriples::Resource equality needs work
+      it 'returns valid item' do
         expect(described_class.all.map(&:rdf_subject))
-          .to contain_exactly provider.rdf_subject
+          .to contain_exactly rdf_subject
       end
-    end
 
-    context 'with bnode provider' do
-      include_context 'with indexed item'
+      it 'assigns :name Providers' do
+        expect(described_class.all.map(&:name))
+          .to include name
+      end
+    end
 
-      let(:provider) { DPLA::MAP::Agent.new }
+    context 'with bnode' do
+      include_context 'bnode indexed in Solr'
 
-      it 'ignores bnodes' do
-        expect(described_class.all).to be_empty
+      it 'ingnores bnode' do
+        expect(described_class.all.count).to eq 0
```

Comment:
Does this no longer return an array that responds to `#empty?` ?  It's still documented as doing so at https://github.com/dpla/KriKri/pull/177/files#diff-c351630f46375b8711be0693562042e8R63